baselayout/{passwd,group}: Add missing entries

krnowak opened the pull request:
Updated netperf needs netperf user and group. Updated systemd needs
various systemd users and groups. Dnsmasq also seems to require its
own user/group.

All this is added to prevent systemd-sysusers adding these to
/etc/passwd. And systemd-sysusers adds these, because the updated
user/group eclass in portage-stable now drops configuration files into
/usr/lib/sysusers.d. Maybe at some point we will switch over to
(patched?) systemd-sysusers, so this catch-up game won't be necessary,
but we are not there yet.

krnowak committed:
baselayout/{passwd,group}: Add missing entries

Updated netperf needs netperf user and group. Updated systemd needs
various systemd users and groups. Dnsmasq also seems to require its
own user/group.

All this is added to prevent systemd-sysusers adding these to
/etc/passwd. And systemd-sysusers adds these, because the updated
user/group eclass in portage-stable now drops configuration files into
/usr/lib/sysusers.d. Maybe at some point we will switch over to
(patched?) systemd-sysusers, so this catch-up game won't be necessary,
but we are not there yet.

krnowak commented:
Hmph, just had a look a bit more. Currently we seem to have `systemd-oom`, `systemd-timesync` and `systemd-coredump` users and groups in `/etc/passwd`, because we didn't have them before in `/usr/share/baselayout/passwd`, so they were created by systemd from the `/usr/lib/sysusers.d/systemd.conf`… These users/groups got assigned ids 999, 998 and 997, respectively.

I can't see if `systemd-oom` and `systemd-coredump` has any files, but `systemd-timesync` surely has some in `/var/lib/systemd/timesync` and in `/run/systemd/timesync`. That would probably mean, that we should hardcode `998` uid/gid for `systemd-timesync`, but it's possible that someone added some more config files to `/usr/lib/sysusers.d` with ignition, so systemd-timesync could get uid/gid like 987 too.

I'm wondering if hardcoding `systemd-timesync` to 195 will be problematic. If you create a new node with Flatcar, systemd-timesync will have uid/gid 195 and that's that. But if we have a node that uses stable channel and just received an update with this change in baselayout, then it's going to have systemd-timesync with uid/gid 998 (or else) in `/etc/passwd` and with uid/gid 195 in `/usr/share/baselayout/passwd`. I'd guess that entries in `/etc/passwd` have precedence over entries in `/usr/share/baselayout/passwd`, so this change should be still backward compatible.

What do you think?

pothos commented:
Yes, `/etc/passwd` takes precedence, let's continue but at least annotate here that users may have 999, 998, 997 already taken now due to this.

krnowak committed:
Add a warning about user/group ranges

Certain ID range could possibly be already used in running systemds,
so be careful not to use it in the future.

krnowak commented:
> Yes, `/etc/passwd` takes precedence, let's continue but at least annotate here that users may have 999, 998, 997 already taken now due to this.
> Edit: Something like `baselayout/baselayout/WARNING`

Good idea, added.
